File: CSharp/Inventory_and_Order_Processing/Models/Order.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Models
{
    public class Order
    {
        public string CustomerName { get; private set; }
        public DateTime Timestamp { get; private set; }
        private List<Product> _products = new List<Product>();

        public Order(string customerName)
        {
            CustomerName = customerName;
            Timestamp = DateTime.Now;
        }

        public void AddProduct(Product product)
        {
            _products.Add(product);
        }

        public double CalculateTotal()
        {
            double total = 0;
            foreach (Product product in _products)
            {
                total += product.Price * product.Quantity;
            }
            return total;
        }

        public void DisplayInfo()
        {
            Console.WriteLine($"Order for: {CustomerName} | Time: {Timestamp}");
            foreach (Product product in _products)
            {
                product.DisplayInfo();
            }
            Console.WriteLine($"Order Total: {CalculateTotal()}");
        }
        public List<Product> GetProducts()
        {
            return _products;
        }
    }
}

File: Python/Inventory_and_Order_Processing/models/order.py
from datetime import datetime

class Order:
    def __init__(self, customer_name):
        self._customer_name = customer_name
        self._products = []
        self._timestamp = datetime.now()
        
    def add_product(self, product):
        self._products.append(product)
        
    def calculate_total(self):
        total = 0
        for product in self._products:
            total += product._price * product._quantity
        return total
    
    def display_info(self):
        print(f"Order for: {self._customer_name} | Time: {self._timestamp}")
        for product in self._products:
            product.display_info()
        print(f"Order Total: {self.calculate_total()}")